code/modules/research/server.dm

```diff
@@ -59,8 +57,9 @@
 			heat_health = min(100, heat_health + 1)
 		if(T0C to (T20C + 20))
 			heat_health = Clamp(heat_health, 0, 100)
-		if((T20C + 20) to (T0C + 70))
-			heat_health = max(0, heat_health - 1)
+		if((T20C + 20) to INFINITY)
+			var/excess = environment.temperature - (T20C + 20)
+			heat_health = max(0, -(excess/40))
 	if(heat_health <= 0)
 		/*griefProtection() This seems to get called twice before running any code that deletes/damages the server or it's files anwyay.
```

Comment:
This should go back in, as this if loop has a chance to remove tech levels.

You might be able to put it in the `if(prob(1))` section, to reduce the number of times it's called.

something along the lines of
```
if(!updateRD)
     griefProtection()
updateRD++
T.level--
```